src/BlogPostPage.jsx

```diff
@@ -87,21 +298,173 @@ if (blog) {
+
+          {/* RENDER BLOG HEADER - DATE */}
+
+          <p>
+            <Widget
+              src="${REPL_ACCOUNT}/widget/TimeAgo"
+              props={{ blockHeight: blog.block_height, blockTimestamp: blog.block_timestamp }}
+            />
+          </p>
+
+          {/* RENDER BLOG HEADER - AUTHOR INFO */}
+
+          <Widget
+            src="${REPL_ACCOUNT}/widget/AccountProfile"
+            key={blog.accountId}
+            props={{
+              accountId: blog.accountId,
+            }}
+          />
+        </TitleSection>
+
+        {/* RENDER BLOG BODY */}
+        {markdownObj.map((element, index) => {
+          if ((index <= 1 && element.type === "header") || element.type === "header-image") {
+            return;
+          } else {
+            return (
+              <ItemWrapper style={{ marginTop: element.type === "list-item" ? "" : "2em" }}>
+                <Widget src="${REPL_ACCOUNT}/widget/SocialMarkdown" props={{ text: element.content }} />
+              </ItemWrapper>
+            );
+          }
+        })}
+
+        <>
+          {state.content && (
+            <Content>
+              {state.content.text && !state.editPost && (
+                <Widget src="${REPL_ACCOUNT}/widget/SocialMarkdown" props={{ text: state.content.text }} />
+              )}
+
+              {state.editPost && (
+                <div className="mb-2">
+                  <Widget
+                    src="${REPL_ACCOUNT}/widget/Posts.Edit"
+                    props={{
+                      item: { accountId, blockHeight },
+                      content: state.content,
+                      onEdit: toggleEdit,
+                    }}
+                  />
+                </div>
+              )}
+
+              {state.content.image && (
+                <Widget
+                  src="${REPL_MOB}/widget/Image"
+                  props={{
+                    image: state.content.image,
+                  }}
+                />
+              )}
+            </Content>
+          )}
+        </>
+      </BlogPostContentWrapper>
+
+      {/* RENDER BLOG FOOTER - COMMENTS / LIKES / ETC */}
+      <BlogPostFooterWrapper>
+        {blockHeight !== "now" && (
+          <Actions>
+            <PromptSignUpWrapper>
+              <Widget
+                src="${REPL_ACCOUNT}/widget/v1.LikeButton"
+                props={{
+                  item,
+                  notifyAccountId,
+                  likes: state.likes,
+                }}
+              />
+
+              <Widget
+                src="${REPL_ACCOUNT}/widget/CommentButton"
+                props={{
+                  item,
+                  onClick: () => State.update({ showReply: !state.showReply }),
+                }}
+              />
+            </PromptSignUpWrapper>
+            <Widget
+              src="${REPL_ACCOUNT}/widget/CopyUrlButton"
+              props={{
+                url: postUrl,
```

Comment:
I'm seeing a null in the UI when I try to copy or share. Looks like `postUrl` is never set

Reply:
ah, that's part of what's in flight. buttoned that up and working on getting the other actions going along with the sign in feature.